src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.Operation;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class OperationStrategyImplTest {
+    private OperationStrategyImpl operationStrategy;
+    private Map<Operation, FruitOperationHandler> handlerMap;
+
+    @BeforeEach
+    public void setUp() {
+        handlerMap = new HashMap<>();
+        FruitOperationHandler balanceHandler = new BalanceHandler();
+        FruitOperationHandler supplyHandler = new SupplyOperation();
+        FruitOperationHandler purchaseHandler = new PurchaseOperation();
+        FruitOperationHandler returnHandler = new ReturnOperation();
+
+        handlerMap.put(Operation.BALANCE, balanceHandler);
+        handlerMap.put(Operation.SUPPLY, supplyHandler);
+        handlerMap.put(Operation.PURCHASE, purchaseHandler);
+        handlerMap.put(Operation.RETURN, returnHandler);
+
+        operationStrategy = new OperationStrategyImpl(handlerMap);
+    }
+
+    @Test
+    public void getHandler_validOperation_returnsCorrectHandler() {
```

Comment:
The test 'getHandler_validOperation_returnsCorrectHandler' is testing all operations in one test. It would be better to split this into separate tests for each operation to make it clearer which operation failed if the test does not pass.